test/functional/test_framework/test_node.py

```diff
@@ -97,15 +98,23 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             "-debugexclude=leveldb",
             "-uacomment=testnode%d" % i,
         ]
+        valgrind_version = self.get_valgrind_version()
```

Comment:
I'd move all this below `if use_valgrind`. Don't forget to throw if `valgrind_version == None`

Reply:
Ok, will do it. Many thanks!